pkg/distributor/distributor.go

```diff
@@ -858,6 +859,100 @@ func (d *Distributor) PushWithCleanup(ctx context.Context, req *mimirpb.WriteReq
 	return &mimirpb.WriteResponse{}, firstPartialErr
 }
 
+// StartBackfill requests the starting of a backfill session.
```

Comment:
[nit] The `distributor.go` file is already huge. Similarly to `query.go`, I would move all the backfill logic to `backfill.go`.

Reply:
Sounds reasonable.